### ISSUE TITLE
x:pending may crash on XQuery

### ISSUE THREAD

**Issue by AirQuick** [xspec/xspec#446]
```xml
	<x:pending label="Pending 1">
		<x:scenario label="Scenario 1">
			<x:call function="false" />
			<x:expect label="Expect 1" select="false()" />
		</x:scenario>
	</x:pending>

	<x:scenario label="Scenario 2" />
```

is terminated on XQuery:

```
  XPST0003: Unexpected token ")" at start of expression
Static error(s) in query

*** Error running the test suite
```

Also

```xml
	<x:pending label="Pending 1">
		<x:scenario label="Scenario 1">
			<x:call function="false" />
			<x:expect label="Expect 1" select="false()" />
		</x:scenario>
	</x:pending>

	<x:pending label="Pending 2">
		<x:scenario label="Scenario 2">
			<x:call function="false" />
			<x:expect label="Expect 2" select="false()" />
		</x:scenario>
	</x:pending>
```

or

```xml
	<x:scenario label="Scenario 1" pending="Pending 1">
		<x:call function="false" />
		<x:expect label="Expect 1" select="false()" />
	</x:scenario>

	<x:pending label="Pending 2">
		<x:scenario label="Scenario 2">
			<x:call function="false" />
			<x:expect label="Expect 2" select="false()" />
		</x:scenario>
	</x:pending>
```

Note: XSLT isn't terminated.

**Comment by galtm** [comment]
The examples above use x:scenario inside x:pending. I came across something similar with x:expect inside x:pending, where the generated XQuery code had static syntax errors.

It looked as if the part of generate-common-tests.xsl that computes the value of $last to pass to the x:output-call template is not accounting for how x:pending alters the hierarchy. I haven't investigated enough to say that <xsl:with-param name="last".../> is necessarily the place to fix this, but I wanted to mention the scope of the bug, in case you weren't aware of that.

Example:

    <x:scenario label="a non-pending scenario">
      <x:call function="my:square">
        <x:param select="3"/>
      </x:call>
      <x:expect label="must execute the test and return Success" select="9"/>
      <x:pending label="(mark an individual expect as pending)">
        <x:expect label="hide" select="9"/>
      </x:pending>
    </x:scenario>

**Comment by AirQuick** [comment]
I wasn't aware of that at all. In fact I haven't investigated this issue at all. Thanks for shedding some light on this. The sibling-walking navigation in XSpec compiler puzzles me so often...

**Comment by AirQuick** [comment]
Fixed by #1222. 

@galtm,
Are you aware of any other crashing cases?

**Comment by galtm** [comment]
> Are you aware of any other crashing cases?

@AirQuick ,
No, everything I know about this bug is in my earlier comment.

**Comment by galtm** [comment]
While reviewing #1222, I found another instance of a bug that is fixed by that PR. It is similar to the tests you included in #1222 but not identical.

```xml
<?xml version="1.0" encoding="UTF-8"?>
<x:description query="x-urn:test:do-nothing" query-at="do-nothing.xqm"
    xmlns:x="http://www.jenitennison.com/xslt/xspec">
    
    <x:scenario label="Scenario 1">
        <x:call function="false" />
        <x:pending label="Pending 1">
            <x:expect label="Expect 1" select="false()" />
        </x:pending>
    </x:scenario>
    <x:pending label="Pending 2">
        <x:scenario label="Scenario 2"/>  
    </x:pending>    
    
</x:description>
```
